src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -986,22 +779,440 @@ public void AnalyzeCommand_LoadConfigurationFile(string configValue, bool defaul
+        {
+            // If no log file is specified, we will convert the console output into a log file
+            bool captureConsoleOutput = string.IsNullOrEmpty(options.OutputFilePath);
+
+            var command = new TestAnalyzeCommand(fileSystem) { _captureConsoleOutput = captureConsoleOutput };
+            command.DefaultPlugInAssemblies = new Assembly[] { typeof(AnalyzeCommandBaseTests).Assembly };
+
+            try
+            {
+                HashUtilities.FileSystem = fileSystem;
+                command.Run(options).Should().Be(expectedReturnCode);
+            }
+            finally
+            {
+                HashUtilities.FileSystem = null;
+            }
+
+            return captureConsoleOutput 
+                ? ConvertConsoleOutputToSarifLog(command._consoleLogger.CapturedOutput)
+                : JsonConvert.DeserializeObject<SarifLog>(File.ReadAllText(options.OutputFilePath));
+        }
+
+        private static SarifLog ConvertConsoleOutputToSarifLog(string consoleOutput)
+        {
+            var sb = new StringBuilder();
+            var converter = new MSBuildConverter(verbose: true);
+
+            using (var input = new MemoryStream(Encoding.UTF8.GetBytes(consoleOutput)))
+            using (var outputTextWriter = new StringWriter(sb))
+            using (var outputJson = new JsonTextWriter(outputTextWriter))
+            using (var output = new ResultLogJsonWriter(outputJson))
+            {
+                converter.Convert(input, output, OptionallyEmittedData.None);
+            }
+            return JsonConvert.DeserializeObject<SarifLog>(sb.ToString());
+        }
+
+        private class ResultsCachingTestCase
+        {
+            private IFileSystem _fileSystem;
+
+            public ResultsCachingTestCase()
+            {
+                ExpectedReturnCode = SUCCESS;
+            }
+
+            public bool Verbose;
+
+            public IList<string> Files;
+
+            public IFileSystem FileSystem
+            {
+                get
+                {
+                    if (_fileSystem == null)
+                    {
+                        _fileSystem = CreateDefaultFileSystemForResultsCaching(Files);
+                    }
+                    return _fileSystem;
+                }
+
+                set => _fileSystem = value;
+            }
+
+            public int ExpectedReturnCode;
+
+            public int ExpectedResultsCount =>
+                // Non-verbose results
+                (ExpectedErrorCount + ExpectedWarningCount) +
+                // Verbose results
+                (Verbose
+                    ? ExpectedNoteCount + ExpectedPassCount + ExpectedInformationalCount +
+                      ExpectedOpenCount + ExpectedReviewCount + ExpectedNotApplicableCount
+                    : 0);
+
+            public int ExpectedErrorCount =>
+                Files.Where((f) => f.Contains("Error")).Count() +
```

Comment:
>Files.Where((f) => f.Contains("Error")).Count() [](start = 16, length = 47)

`Count` takes a predicate:
```
Files.Count(f => f.Contains("Error")) +
```

Reply:
And having said that, this code is repeated 9 times, so perhaps a helper:
```
private int NumberOfFilesWithNameLike(string s) => Files.Count(f => f.Contains(s));
```


---
In reply to: [322905337](https://github.com/microsoft/sarif-sdk/pull/1670#discussion_r322905337) [](ancestors = 322905337)